buildpacks/ruby/src/layers/bundle_install_layer.rs

```diff
@@ -53,211 +36,103 @@ pub(crate) fn call(
-            }
         }
         LayerState::Empty { cause } => match cause {
-            EmptyLayerCause::NewlyCreated => InstallState::Run(String::new()),
+            EmptyLayerCause::NewlyCreated => (),
             EmptyLayerCause::InvalidMetadataAction { cause }
             | EmptyLayerCause::RestoredLayerAction { cause } => {
                 print::sub_bullet(cause);
-                InstallState::Run(String::new())
             }
         },
-    };
+    }
 
     let env = {
         let layer_env = layer_env(&layer_ref.path(), &context.app_dir, without);
         layer_ref.write_env(&layer_env)?;
         layer_env.apply(Scope::Build, env)
     };
 
-    match install_state {
-        InstallState::Run(reason) => {
-            if !reason.is_empty() {
-                print::sub_bullet(reason);
-            }
+    let mut cmd = Command::new("bundle");
+    cmd.arg("install")
+        .env_clear() // Current process env vars already merged into env
+        .envs(&env);
 
-            let mut cmd = Command::new("bundle");
-            cmd.arg("install")
-                .env_clear() // Current process env vars already merged into env
-                .envs(&env);
-
-            print::sub_stream_cmd(cmd.named_fn(|cmd| display_name(cmd, &env)))
-                .map_err(|error| {
-                    fun_run::map_which_problem(error, cmd.mut_cmd(), env.get("PATH").cloned())
-                })
-                .map_err(RubyBuildpackError::BundleInstallCommandError)?;
-
-            print::sub_time_cmd(
-                Command::new("bundle")
-                    .args(["clean", "--force"])
-                    .env_clear()
-                    .envs(&env),
-            )
-            .map_err(RubyBuildpackError::BundleInstallCommandError)?;
-        }
-        InstallState::Skip(checked) => {
-            let bundle_install = style::value("bundle install");
-            let help = style::important("HELP");
-
-            print::sub_bullet(format!(
-                "Skipping {bundle_install} (no changes found in {sources})",
-                sources = SentenceList::new(&checked).join_str("or")
-            ));
-            print::sub_bullet(format!(
-                "{help} To force run {bundle_install} set {}",
-                style::value(format!("{SKIP_DIGEST_ENV_KEY}=1"))
-            ));
-        }
-    }
+    print::sub_stream_cmd(cmd.named_fn(|cmd| display_name(cmd, &env)))
+        .map_err(|error| fun_run::map_which_problem(error, cmd.mut_cmd(), env.get("PATH").cloned()))
+        .map_err(RubyBuildpackError::BundleInstallCommandError)?;
+
+    print::sub_time_cmd(
+        Command::new("bundle")
+            .args(["clean", "--force"])
+            .env_clear()
+            .envs(&env),
+    )
+    .map_err(RubyBuildpackError::BundleInstallCommandError)?;
 
     layer_ref.read_env()
 }
 
-pub(crate) type Metadata = MetadataV3;
+pub(crate) type Metadata = MetadataV4;
```

Comment:
Small nitpick - I think this type alias could be named `CurrentMetadataVersion`,  `LatestMetadata`, or something to that affect. Only because my first instinct when reading some of the function signatures was that `Metadata` might be the first struct you started with (i.e.; V0) before realizing you'd need subsequent versions.

Reply:
I'm not sure how much it helps, but I moved the type alias to the top of the file and added a comment on it. I'm hesitant to apply an adjective here.